### PR TITLE
feat(arc-saas): frontend components with backend integration

### PR DESCRIPTION
**Pull request by Deepika516**
## Description
frontend components with backend integration

Fixes #34 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/saas-ui/src/app/on-boarding/commands/add-tenant-lead.command.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(
+    apiService: ApiService,
+    adapter: IAdapter<T>,
+    leadId: string,
+    @Inject(APP_CONFIG) private readonly appConfig?: IAnyObject,
+  ) {
+    super(
+      apiService,
+      adapter,
+      `${appConfig.tenantMgmtFacadeUrl}/leads/${leadId}/tenants`,
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/commands/add-tenant-lead.command.ts:24]: here facadeUrl is used 
somewhere above service url was used why this ambiguity

File: projects/saas-ui/src/app/on-boarding/on-boarding-service.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    private readonly validateLeadAdapter: ValidateLeadAdapter,
+    private readonly getPlanAdapter: GetPlanAdapter,
+  ) {}
+
+  getValidateLead(): Subject<Lead> {
+    return this.validateLead$;
+  }
+  setValidateLead(lead: Lead) {
+    this.validateLead$.next(lead);
+  }
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/on-boarding-service.ts:56]: where is this used, cant find its reference

Comment (yeshamavani) [reply projects/saas-ui/src/app/on-boarding/on-boarding-service.ts:56]: also check other unused code and remove

File: projects/saas-ui/src/environment.ts
Comment (yeshamavani) [inline projects/saas-ui/src/environment.ts:1]: dont push with values 
just commit a sample file here

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import {Tenant} from '../model/lead-tenant.model';
+import {Observable} from 'rxjs';
+
+export class AddTenantCommand<T> extends PostAPICommand<T> {
+  parameters: {
+    data: Tenant;
+  };
+  execute(): Observable<T> {
+    throw new Error('Method not implemented.');
+  }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts:15]: removed

File: projects/saas-ui/src/app/shared /auth/auth-routing.module.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/shared /auth/auth-routing.module.ts:1]: Done

File: projects/saas-ui/src/app/shared /auth/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        ],
+      ],
+    });
+  }
+
+  showPassword = false;
+
+  onSubmit() {
+    // TOdo
+  }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/shared /auth/login/login.component.ts:47]: removed

File: projects/saas-ui/src/environment.ts
Comment (Deepika516) [reply projects/saas-ui/src/environment.ts:1]: Done

File: projects/arc-lib/src/lib/core/auth/auth.service.ts [hunk 5]
```diff
@@ -334,6 +334,48 @@ export class AuthService {
...
+    clientId.type = 'hidden';
+    clientId.name = 'client_id';
+    clientId.value = 'test_client_id';
+    form.appendChild(clientId);
+
+    const clientSecret = document.createElement('input');
+    clientSecret.type = 'hidden';
+    clientSecret.name = 'client_secret';
+    clientSecret.value = 'test_client_secret';
+    form.appendChild(clientSecret);
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/auth.service.ts:353]: done

File: projects/saas-ui/src/app/main/enums/tenant-status.enum.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export enum TenantStatus {
+  ACTIVE = 0,
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/enums/tenant-status.enum.ts:2]: done

File: projects/saas-ui/src/assets/images/Illustration.svg
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/assets/images/Illustration.svg:1]: done

File: projects/saas-ui/src/app/main/lead-list/commands/delete-tenant.command.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+import {Inject} from '@angular/core';
+import {Observable} from 'rxjs';
+
+export class DeleteTenantCommand<T> extends DelAPICommand<T> {
+  execute(): Observable<T> {
+    throw new Error('Method not implemented.');
+  }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/commands/delete-tenant.command.ts:11]: done

File: projects/saas-ui/src/app/main/lead-list/commands/edit-tenant.command.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+import {Inject} from '@angular/core';
+import {Tenant} from '../model/lead-tenant.model';
+import {Observable} from 'rxjs';
+
+export class EditTenantCommand<T> extends PatchAPICommand<T> {
+  data: Tenant;
+  execute(): Observable<T> {
+    throw new Error('Method not implemented.');
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/commands/edit-tenant.command.ts:11]: ok

File: projects/arc-lib/src/lib/core/auth/auth.service.ts [hunk 9]
```diff
@@ -334,6 +334,48 @@ export class AuthService {
...
+    clientId.name = 'client_id';
+    clientId.value = '3hcd79ut813reesb7mvlomnbrv';
+    form.appendChild(clientId);
+
+    const clientSecret = document.createElement('input');
+    clientSecret.type = 'hidden';
+    clientSecret.name = 'logout_uri';
+    clientSecret.value = 'http://localhost:4200/auth/login';
+    form.appendChild(clientSecret);
+
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/auth.service.ts:375]: ok done

File: projects/saas-ui/src/app/main/lead-list/model/lead-tenant.model.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/model/lead-tenant.model.ts:1]: Done

File: projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    this.getOnBoardingTenants();
+  }
+
+  getOnBoardingTenants() {
+    this.onboardingService
+      .getPlanOptions(null, null, this.filter)
+      .pipe(takeUntil(this._destroy$))
+      .subscribe(res => {
+        this.rowData = res.map(item => {
+          console.log(res[0]);
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts:55]: done

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+      state: ['', Validators.required],
+      city: ['', Validators.required],
+      zip: ['', Validators.required],
+    });
+  }
+
+  onSubmit() {
+    if (this.addLeadForm.valid) {
+      const userData = this.addLeadForm.value;
+      console.log(userData);
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.ts:42]: removed

File: projects/saas-ui/src/app/on-boarding/commands/verify-email.command.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IApiService, IAdapter, PostAPICommand} from '@project-lib/core/api';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+import {environment} from 'projects/saas-ui/src/environment';
+import {Observable} from 'rxjs';
+
+export class VerifyEmailCommand<T> extends PostAPICommand<T> {
+  parameters: any;
+  execute(): Observable<T> {
+    throw new Error('Method not implemented.');
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/commands/verify-email.command.ts:11]: removed

File: projects/saas-ui/src/app/on-boarding/on-boarding-service.ts (hunk 2 above)
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/on-boarding-service.ts:56]: removed

File: projects/saas-ui/src/app/on-boarding/on-boarding-routing.module.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import {NgModule} from '@angular/core';
+import {RouterModule, Routes} from '@angular/router';
+import {AddLeadComponent} from './add-lead/add-lead.component';
+import {OnBoardingComponent} from './on-boarding.component';
+import {EmailVerificationComponent} from './email-verification/email-verification.component';
+import {VerificationCompleteComponent} from './verification-complete/verification-complete.component';
+import {EmailVerifyGuard} from './guards';
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/on-boarding-routing.module.ts:7]: done

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead-routing.module.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/add-lead/add-lead-routing.module.ts:1]: i wanted to used lazy loading thats y i created that

File: projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts:1]: we were using ValidateLeadAdapter in onBoarding Service, still i replaced that with anyAdapter

File: projects/saas-ui/src/app/main/lead-list/commands/get-tenant-by-id.command.ts [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import {ApiService, GetAPICommand, IAdapter} from '@project-lib/core/api';
+
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+import {Inject} from '@angular/core';
+import {Observable} from 'rxjs';
+
+export class GetTenantByIdCommand<T> extends GetAPICommand<T> {
+  execute(): Observable<T> {
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/commands/get-tenant-by-id.command.ts:9]: removed

File: projects/saas-ui/src/app/shared/app.module.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared/app.module.ts:1]: why app.module again in shared folder ?

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-lib/src/lib/core/auth/commands/get-current-user.comand.ts [hunk 15]
```diff
@@ -1,11 +1,13 @@
...
+import {Inject} from '@angular/core';
+import {APP_CONFIG} from '@project-lib/app-config';
+import {GetAPICommand, IAdapter, IApiService} from '@project-lib/core/api';
 
 export class GetCurrentUserCommand<T> extends GetAPICommand<T> {
   constructor(
     apiService: IApiService,
     adapter: IAdapter<T>,
-    appConfig: IAnyObject,
+    @Inject(APP_CONFIG) private readonly appConfig: IAnyObject,
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/arc-lib/src/lib/core/auth/commands/get-current-user.comand.ts:10]: Done

File: projects/arc-lib/src/lib/core/auth/commands/get-token.command.ts [hunk 16]
```diff
@@ -1,7 +1,14 @@
...
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAdapter, IApiService, PostAPICommand} from '@project-lib/core/api';
+import {IAnyObject} from '@project-lib/core/i-any-object';
 
 export class GetTokenCommand<T> extends PostAPICommand<T> {
-  constructor(apiService: IApiService, adapter: IAdapter<T>, appConfig: any) {
+  constructor(
+    apiService: IApiService,
+    adapter: IAdapter<T>,
+    @Inject(APP_CONFIG) private readonly appConfig?: IAnyObject,
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/arc-lib/src/lib/core/auth/commands/get-token.command.ts:10]: Done

File: projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+import {Leadtenant} from '../../model/lead-tenant.model';
+
+export class AddTenantCommand<T> extends PostAPICommand<T> {
+  parameters: {
+    data: Leadtenant;
+  };
+  constructor(
+    apiService: ApiService,
+    adapter: IAdapter<T>,
+    @Inject(APP_CONFIG) private readonly appConfig: IAnyObject,
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts:15]: Done

File: projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+    private fb: FormBuilder,
+  ) {
+    super(route, location);
+  }
+
+  ngOnInit(): void {
+    this.getOnBoardingTenants();
+  }
+
+  getOnBoardingTenants() {
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts:49]: Done

File: projects/arc-lib/src/lib/core/auth/commands/get-current-user.comand.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/arc-lib/src/lib/core/auth/commands/get-current-user.comand.ts:10]: Done with Complete testing of the commands for the components in this PR.

File: projects/arc-lib/src/lib/core/auth/commands/get-token.command.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/arc-lib/src/lib/core/auth/commands/get-token.command.ts:10]: Done

File: projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts (hunk 17 above)
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/main/lead-list/commands/add-tenant.command.ts:15]: Done

File: projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/main/manage-plans/manage-plans.component.ts:49]: Changed to getPlans()

File: projects/saas-ui/src/app/main/model/lead-tenant.model.ts
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/main/model/lead-tenant.model.ts:1]: Removed leadTenant

File: projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts:1]: ok will take care , but i already did the changes ,I removed 
ValidateLeadAdapter

File: projects/arc-lib/src/lib/core/auth/auth.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/auth.service.ts:362]: changed

File: projects/saas-ui/src/app/main/lead-list/model/tenant.model.ts
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/lead-list/model/tenant.model.ts:1]: removed

File: projects/saas-ui/src/app/shared/app.module.ts
Comment (Surbhi-sharma1) [reply projects/saas-ui/src/app/shared/app.module.ts:1]: Removed..

File: projects/saas-ui/src/app/on-boarding/on-boarding-service.ts [hunk 19]
```diff
@@ -0,0 +1,156 @@
...
+
+  public validateEmail(
+    code: string,
+    leadId: string,
+  ): Observable<{leadId: string; token: string}> {
+    const command: VerifyEmailCommand<{leadId: string; token: string}> =
+      new VerifyEmailCommand(this.apiService, this.validateLeadAdapter, leadId);
+    // sonarignore:end
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/on-boarding-service.ts]: can see it is being used here

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/saas-ui/src/app/main/main.component.ts [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+      .onItemClick()
+      .pipe(takeUntil(this._destroy$))
+      .subscribe(menu => {
+        if (menu.tag === 'userMenu' && menu.item.data === 'logout') {
+          this.authService
+            .logout()
+            .pipe(concatMap(async () => await this.authService.logoutCognito()))
+            .subscribe();
+          console.log('login works');
+        }
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/saas-ui/src/app/main/main.component.ts:106]: ??

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead-routing.module.ts
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/saas-ui/src/app/on-boarding/add-lead/add-lead-routing.module.ts:1]: this is anywhich ways not lazyloading 
its routing and not appropriate

**Comment by Surbhi-sharma1** [comment]
https://github.com/sourcefuse/angular-boilerplate/pull/39

**Pull request merged**